trikScriptRunner/src/scriptEngineWorker.cpp

```diff
@@ -48,31 +49,48 @@ using namespace trikScriptRunner;
 using namespace trikControl;
 using namespace trikNetwork;
 
-Q_DECLARE_METATYPE(BatteryInterface*)
-Q_DECLARE_METATYPE(ColorSensorInterface*)
-Q_DECLARE_METATYPE(FifoInterface*)
-Q_DECLARE_METATYPE(DisplayInterface*)
-Q_DECLARE_METATYPE(EncoderInterface*)
-Q_DECLARE_METATYPE(EventCodeInterface*)
-Q_DECLARE_METATYPE(EventDeviceInterface*)
-Q_DECLARE_METATYPE(EventInterface*)
-Q_DECLARE_METATYPE(GamepadInterface*)
-Q_DECLARE_METATYPE(GyroSensorInterface*)
-Q_DECLARE_METATYPE(KeysInterface*)
-Q_DECLARE_METATYPE(LedInterface*)
-Q_DECLARE_METATYPE(LineSensorInterface*)
-Q_DECLARE_METATYPE(MailboxInterface*)
-Q_DECLARE_METATYPE(MarkerInterface*)
-Q_DECLARE_METATYPE(MotorInterface*)
-Q_DECLARE_METATYPE(ObjectSensorInterface*)
-Q_DECLARE_METATYPE(SoundSensorInterface*)
-Q_DECLARE_METATYPE(SensorInterface*)
-Q_DECLARE_METATYPE(Threading*)
-Q_DECLARE_METATYPE(VectorSensorInterface*)
 Q_DECLARE_METATYPE(QVector<int>)
 Q_DECLARE_METATYPE(trikKernel::TimeVal)
 Q_DECLARE_METATYPE(QTimer*)
 
+#define DECLARE_METATYPE_TEMPLATE(TYPE) \
+	Q_DECLARE_METATYPE(TYPE*)
+
+#define REGISTER_METATYPE_FOR_ENGINE(TYPE) \
+	Scriptable<TYPE>::registerMetatype(engine);
+
+#define REGISTER_METATYPE(TYPE) \
+	qRegisterMetaType<TYPE*>(TYPE::staticMetaObject.className());
+
+/// Here we define a comfortablle template that registers all devices used in trik.
```

Comment:
'comfortable' is an adjective for my chair. Here you probably mean 'convenient' or 'reusable'. As for me, this whole comment should be like "ATTENTION: do not forget to append newly created device to this list!"